File: algorithms/100-199/198. House Robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if(nums.empty()) return 0;
        int n = nums.size();
        if (n == 1) return nums[0];
        int prev = nums[0], curr = max(nums[0], nums[1]);
        int temp;
        for (int i = 2; i < n; ++i){
            temp = curr;
            curr = max(prev + nums[i], temp);
            prev = temp;
        }
        return curr;
    }
};

File: algorithms/100-199/198. House Robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if not nums:
            return 0
        if len(nums) <= 2:
            return max(nums)
        prev, curr = nums[0], max(nums[0], nums[1])
        for i in range(2, len(nums)):
            curr, prev = max(prev + nums[i], curr), curr
        return curr
            
            